### PR TITLE
GEOS-4972 Allow the link attribute in a GeoRSS feed to be templated like the title and description attributes

### PR DESCRIPTION
**Pull request by kc7bfi**
This allows the link to be templated in GeoRSS feeds

### CODE REVIEW

File: src/wms/src/main/java/org/geoserver/wms/georss/AtomUtils.java [hunk 1]
```diff
@@ -86,18 +86,22 @@ public static String dateToRFC3339(Date d){
...
-        params.put("format", "application/atom+xml");
-        params.put("layers",  nsPrefix + ":" + feature.getType().getTypeName());
-        params.put("featureid", feature.getID());
-        
-        return ResponseUtils.buildURL(context.getRequest().getBaseUrl(),
-             "wms/reflect",
-             params,
-             URLType.SERVICE);
+		try {
+			return featureTemplate.link(feature);
```
Comment (aaime) [inline src/wms/src/main/java/org/geoserver/wms/georss/AtomUtils.java]: Not respecting coding conventions. Please check and follow the guidelines for pull requests: https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md#commit-guidelines

Also check the rest of the document (for example, have you already sent a contribution agreement?, did the files need to have copyright headers updated?)

**Comment by aaime** [comment]
The change is in line with the rest of the templating engine, thanks for that, there are visible formatting issues though. 
I was about to write "the documentation should be updated too" but it seems we have none on the template customization to start with. Not a requirement, but we'll love if you could write some docs too.

**Comment by kc7bfi** [comment]
If you can tell me where in the documentation I should create a page for GeoRSS templating I could add a page.

**Comment by aaime** [comment]
I believe docs would have to go in this page:
http://docs.geoserver.org/latest/en/user/tutorials/georss/georss.html
that is, source wise, here:
https://github.com/geoserver/geoserver/blob/master/doc/en/user/source/tutorials/georss/georss.rst

**Comment by aaime** [comment]
If you check the tutorials, there are pages talking about templates already that you can link/take inspiration from

**Comment by kc7bfi** [comment]
GEOS-4972 Allow the link attribute in a GeoRSS feed to be templated like the title and description attributes

**Comment by aaime** [comment]
Squashed into a single commit and manually merged here: https://github.com/geoserver/geoserver/commit/4a9e065d6e98cd8a83d7fdb787d492ab25889e7a

**Comment by aaime** [comment]
To clarify, the changes are now merged on master. Since this is a new feature, you will be able to ask for a backport on the developer list in a month from today (assuming the patch does not cause any trouble)

**Pull request closed without merging**